Code/Editor/EditorViewportWidget.h

```diff
@@ -65,130 +66,120 @@ namespace AzToolsFramework
+    , private IEditorNotifyListener
+    , private IUndoManagerListener
+    , private Camera::EditorCameraRequestBus::Handler
+    , private Camera::CameraNotificationBus::Handler
+    , private AzFramework::InputSystemCursorConstraintRequestBus::Handler
+    , private AzToolsFramework::ViewportInteraction::ViewportFreezeRequestBus::Handler
+    , private AzToolsFramework::ViewportInteraction::MainEditorViewportInteractionRequestBus::Handler
+    , private AzFramework::AssetCatalogEventBus::Handler
+    , private AZ::RPI::SceneNotificationBus::Handler
 {
 AZ_POP_DISABLE_DLL_EXPORT_MEMBER_WARNING
 AZ_POP_DISABLE_DLL_EXPORT_BASECLASS_WARNING
     Q_OBJECT
-public:
-    struct SResolution
-    {
-        SResolution()
-            : width(0)
-            , height(0)
-        {
-        }
-
-        SResolution(int w, int h)
-            : width(w)
-            , height(h)
-        {
-        }
-
-        int width;
-        int height;
-    };
 
 public:
     EditorViewportWidget(const QString& name, QWidget* parent = nullptr);
+    ~EditorViewportWidget() override;
 
     static const GUID& GetClassID()
     {
         return QtViewport::GetClassID<EditorViewportWidget>();
     }
 
-    /** Get type of this viewport.
-    */
-    virtual EViewportType GetType() const { return ET_ViewportCamera; }
-    virtual void SetType([[maybe_unused]] EViewportType type) { assert(type == ET_ViewportCamera); };
-
-    virtual ~EditorViewportWidget();
-
-    Q_INVOKABLE void InjectFakeMouseMove(int deltaX, int deltaY, Qt::MouseButtons buttons);
-
-    // Replacement for still used CRenderer methods
-    void UnProjectFromScreen(float sx, float sy, float sz, float* px, float* py, float* pz) const;
-    void ProjectToScreen(float ptx, float pty, float ptz, float* sx, float* sy, float* sz) const;
+    static EditorViewportWidget* GetPrimaryViewport();
 
-public:
-    virtual void Update();
+    // Used by ViewPan in some circumstances
+    void ConnectViewportInteractionRequestBus();
+    void DisconnectViewportInteractionRequestBus();
 
-    virtual void ResetContent();
-    virtual void UpdateContent(int flags);
+    // QtViewport/IDisplayViewport/CViewport
+    // These methods are made public in the derived class because they are called with an object whose static type is known to be this class type.
+    void SetFOV(float fov) override;
+    float GetFOV() const override;
 
-    void OnTitleMenu(QMenu* menu) override;
+private:
+    ////////////////////////////////////////////////////////////////////////
+    // Private types ...
 
-    void SetCamera(const CCamera& camera);
-    const CCamera& GetCamera() const { return m_Camera; };
-    virtual void SetViewTM(const Matrix34& tm)
+    enum class ViewSourceType
     {
-        if (m_viewSourceType == ViewSourceType::None)
-        {
-            m_defaultViewTM = tm;
-        }
-        SetViewTM(tm, false);
-    }
+        None,
+        CameraComponent,
+        ViewSourceTypesCount,
```

Comment:
nit/suggestion: I'm not a fan of adding this count member to the enum as it can break the validity of states the enum can be in - it is totally useful but it might be better to just use a constant when looping instead

Reply:
This is legacy stuff, tried not to touch to it too much as I didn't know what it was doing. After this refactor, it's more apparent. This whole enum could probably now be removed - it seems like it's enough to check whether `m_viewEntityId` is valid.

Actually, I would love to refactor this even more. In particular, instead of handling a "component camera" and a "default camera" differently, we could simply create a new entity on viewport creation inside of EditorViewportWidget, add a Camera component and Transform component to that entity, and let it be the "editor camera".

Reply:
Agreed, yeah that sounds much nicer, might be something for another PR one of us can do, thanks for the added context :+1: